dvc/ignore.py

```diff
@@ -66,7 +66,7 @@ def __init__(self, root_dir):
         self._update(root_dir)
         for root, dirs, _ in os.walk(root_dir):
             for d in dirs:
-                self._update(os.path.join(root, d))
+                self._update(os.path.join(root, decode(d, "utf-8")))
```

Comment:
Won't it blowup the same way in other commands? Does repro work? status? import?

Reply:
How are we getting encoded d here in the first place? Is `root_dir` encoded? If so, where did it come from? It feels like the cause is somewhere up there, around argparse or something.

Reply:
@efiop , none of them work, it will require more than half a day to test and investigate the deepness of this one

Reply:
@efiop , we should always received the information _unencoded_, right?
Let me see if there's something up there

Reply:
> none of them work, it will require more than half a day to test and investigate the deepness of this one

But in that case what is the point of fixing these two lines, if everything else will blowup? I feel like you are treating the symptoms and not the cause. Also, why does it work on py3 and not on py2? Maybe argparse is returning us encoded arguments on py2 and not on py3?

Reply:
So, we are receiving the information already _encoded_ (`'\xe8\xb2\x93'`) since `dvc/command/add.py`

Reply:
@efiop , the problem is when trying to do operations with non-unicode filenames, for example: `os.path.join('\xe8', u'derp')` will break, while `os.path.join(u'\xe8', u'derp') will work like charm.

We have the `from __future__ import unicode_literals` thingy that makes all the strings _unicode_ by default, but the parsed argument from `argparse` is a plain `string`, not a `unicode`.

Need to check if there's a way to return unicode from argparse

Reply:
@mroutis sounds good!

Reply:
`type=lambda s: s.decode('utf8')` on each `add_argument` statement that deals with unicode

Reply:
@efiop , I did some changes regarding this, but I can't pin point every single place where strings are not being decoded correctly into unicode; I'm not sure if there's a patch that doesn't involve the whole codebase.

Reply:
Should we spend more time on this one?